decoder.rb

```diff
@@ -1,43 +1,41 @@
-# a hash that contains the key/value pairs of the international morse code
-# alphabet. The key is the morse code character and the value is
-# the corresponding letter.
-#
+def decode_char(char)
+  dictionary = {
+    '.-' => 'A', '-...' => 'B', '-.-.' => 'C', '-..' => 'D',
+    '.' => 'E', '..-.' => 'F', '--.' => 'G',
+    '....' => 'H', '..' => 'I', '.---' => 'J',
+    '-.-' => 'K', '.-..' => 'L', '--' => 'M',
+    '-.' => 'N', '---' => 'O', '.--.' => 'P',
+    '--.-' => 'Q', '.-.' => 'R', '...' => 'S',
+    '-' => 'T', '..-' => 'U', '...-' => 'V',
+    '.--' => 'W', '-..-' => 'X', '-.--' => 'Y',
+    '--..' => 'Z', '.----' => '1', '..---' => '2',
+    '...--' => '3', '....-' => '4', '.....' => '5',
+    '-....' => '6',
+    '--...' => '7',
+    '---..' => '8',
+    '----.' => '9',
+    '-----' => '0'
+  }
+  dictionary[char]
+end
 
-dectionary = {
-  ".-" => "A",
-  "-..." => "B",
-  "-.-." => "C",
-  "-.." => "D",
-  "." => "E",
-  "..-." => "F",
-  "--." => "G",
-  "...." => "H",
-  ".." => "I",
-  ".---" => "J",
-  "-.-" => "K",
-  ".-.." => "L",
-  "--" => "M",
-  "-." => "N",
-  "---" => "O",
-  ".--." => "P",
-  "--.-" => "Q",
-  ".-." => "R",
-  "..." => "S",
-  "-" => "T",
-  "..-" => "U",
-  "...-" => "V",
-  ".--" => "W",
-  "-..-" => "X",
-  "-.--" => "Y",
-  "--.." => "Z",
-  ".----" => "1",
-  "..---" => "2",
-  "...--" => "3",
-  "....-" => "4",
-  "....." => "5",
-  "-...." => "6",
-  "--..." => "7",
-  "---.." => "8",
-  "----." => "9",
-  "-----" => "0",
-}
+def decode_word(word)
+  result = ''
+  word.split.each do |char|
+    result.concat(decode_char(char))
+  end
+  result
+end
+
+# puts decode_word('-- -.--')
```

Comment:
Great job team 🤝

- [ ] Kindly remove commented code from your project as this is regarded as bad practice.